public/js/utils/source-search.js

```diff
@@ -32,6 +33,31 @@ function getSearchCursor(cm, query, pos) {
     typeof query == "string" && query == query.toLowerCase());
 }
 
+function searchOverlay(query) {
+  query = query.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```

Comment:
Had a look into an example of how code mirror does highlighting of text. Basically Codemirror's `addOverlay` function takes a mode object. The mode object should have a token function which is applied as each token in the file is parsed. https://codemirror.net/doc/manual.html#option_mode

The regex just escapes  any special characters in the input